subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeSourceCompileTask.java

```diff
@@ -43,6 +49,12 @@ public boolean isSatisfiedBy(Task element) {
                 return getPreCompiledHeader() != null;
             }
         });
+        getOutputs().doNotCacheIf("Could not determine compiler version", new Spec<Task>() {
+            @Override
+            public boolean isSatisfiedBy(Task element) {
+                return getCompilerIdentifier() == null || VersionNumber.UNKNOWN.equals(getCompilerIdentifier().getVersion());
```

Comment:
`getCompilerIdentifier()` should only be called once.

Reply:
Now I am only calling it once.